site/src/lib/api/blocks/npm.ts

```diff
@@ -0,0 +1,278 @@
+  );
+
+  // move the readme into the block's source folder, if it exists
+  const readmePath = glob.sync("README.md", {
+    cwd: packageFolder,
+    nocase: true,
+  })[0];
+  if (readmePath) {
+    fs.renameSync(
+      path.resolve(packageFolder, readmePath),
+      path.resolve(blockSourceFolder, path.basename(readmePath)),
+    );
+  }
+
+  // check if we have 'example-graph.json', which we then construct a URL for in the extended matadata
+  const includesExampleGraph = await fs.pathExists(
+    path.resolve(blockSourceFolder, "example-graph.json"),
+  );
+
+  // check block-metadata.json contains required properties
+  for (const key of ["blockType", "protocol", "schema", "source", "version"]) {
+    if (!metadataJson[key]) {
+      throw new Error(`block-metadata.json must contain a '${key}' property`, {
+        cause: { code: "INVALID_PACKAGE_CONTENTS" },
+      });
+    }
+  }
+
+  // check that the source file actually exists
+  const sourcePath = metadataJson.source;
+  const sourceFileExists = await fs.pathExists(
+    path.resolve(blockSourceFolder, sourcePath),
+  );
+
+  if (!sourceFileExists) {
+    throw new Error(
+      `block-metadata.json 'source' path '${sourcePath}' does not exist`,
+      {
+        cause: { code: "INVALID_PACKAGE_CONTENTS" },
+      },
+    );
+  }
+
+  // check that the schema actually exists
+  const schemaPath = metadataJson.schema;
+  if (schemaPath.startsWith("http")) {
+    await fetch(schemaPath, { method: "HEAD" });
+  } else {
+    const schemaFileExists = await fs.pathExists(
+      path.resolve(blockSourceFolder, schemaPath),
+    );
+    if (!schemaFileExists) {
+      throw new Error(
+        `block-metadata.json 'schema' path '${schemaPath}' does not exist`,
+        {
+          cause: { code: "INVALID_PACKAGE_CONTENTS" },
+        },
+      );
+    }
+  }
+
+  const now = new Date().toISOString();
+
+  // for blocks developed locally, add a prefix to the storage URL - the R2 bucket is shared across all dev environments
+  let storageNamespacePrefix = "";
+  if (!isRunningOnVercel) {
+    const exec = promisify(child_process.exec);
+
+    const gitConfigUserNameResult = await exec("git config --get user.name");
+    const userName = gitConfigUserNameResult.stdout.trim();
+    storageNamespacePrefix = `local-dev/${slugify(userName, {
+      lower: true,
+      strict: true,
+    })}/`;
+  }
+
+  /**
+   * In future we will store each version in its own folder, and add the version to the folder path
+   * @see https://app.asana.com/0/0/1202539910143057/f (internal)
+   */
+  const remoteStoragePrefix = `${storageNamespacePrefix}${stripLeadingAt(
+    pathWithNamespace,
+  )}`;
+  const publicPackagePath = `${publicBaseR2Url}/${remoteStoragePrefix}`;
+
+  const sourceInformation = {
+    blockDistributionFolderUrl: publicPackagePath,
+    npmPackageName,
+    pathWithNamespace,
+    repository:
+      typeof packageJson.repository === "object" && !!packageJson.repository
```

Comment:
`@todo` – repository may also be a string

Reply:
https://github.com/blockprotocol/blockprotocol/pull/522/commits/360a52e224df3091b4e3f8cca0098a1a7f75421a